src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -208,6 +292,82 @@ public static bool TryParseInt64(string value, out long result)
             return long.TryParse(value, NumberStyles.None, NumberFormatInfo.InvariantInfo, out result);
         }
 
+        internal static unsafe bool TryParseInt32(StringSegment value, out int result)
```

Comment:
Ok then, so TryParseInt64 is public because you're using it from Kestrel then? There should be doc comments on the public one explaining the limitations.